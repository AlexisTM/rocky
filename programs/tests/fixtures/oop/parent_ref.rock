Pasadena looks like nothing
 leave takes a moment
  give back 1
 (eom)
 stay takes a week
  give back 7
 (eom)
(eoc)
Los Angeles looks like Pasadena
 leave takes a moment
  let a day be leave on parent taking a moment
  give back a day plus 1
 (eom)
(eoc)
Tokyo looks like Los Angeles
 leave takes a moment
  let a day be leave on parent taking a moment
  give back a day plus 1
 (eom)
 stay takes forever
  let a day be stay on parent taking forever
  give back a day plus 1
 (eom)
(eoc)

a town wants to be Pasadena
a city wants to be Los Angeles
a metropolis wants to be Tokyo

Say "Parent reference"
say leave at a town taking nothing
say leave at a city taking nothing
say leave at a metropolis taking nothing

Say "Parent reference with grandparent implementation"
say stay at a town taking nothing
say stay at a city taking nothing
say stay at a metropolis taking nothing


