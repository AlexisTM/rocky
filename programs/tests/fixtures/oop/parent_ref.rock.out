Parent reference
1
2
3
Parent reference with grandparent implementation
7
7
8
